frontend/dockerfile/parser/split_command.go

```diff
@@ -37,10 +37,12 @@ func extractBuilderFlags(line string) (string, []string, error) {
 
 	words := []string{}
 	phase := inSpaces
-	word := ""
+	var word strings.Builder
 	quote := '\000'
 	blankOK := false
 	var ch rune
+	parsingFlagArg := false
+	needFlagArg := true
```

Comment:
These seem to be adding an extra dimension (or two?) to the existing `phase` construct, which results in some fairly complex conditionals and state retention across different bits of the code below. Perhaps instead a new state (or states) should be added to cover the flag argument parsing phase?

Reply:
I also thought of adding more states to `phase`. The problem again is that it would significantly complicate the code and the understanding of it, because there would be multiple `inWord` and `inSpaces` variants which would require even more logical ANDs/ORs. Possibly a similar reason why `blankOK` is not encapsulated in `phase`.